client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -0,0 +1,323 @@
+package taskrunner
```

Comment:
I do not like how hard it is to unit test plugin hooks :( - When we get some basic mounting in place I'll cover the majority of this with an E2E test and setup nightly on the `f-csi-plugins` branch.

Reply:
You can test them by instantiating them and poking at methods manually to unit test their functionality.

I'd include at least 1 TaskRunner or E2E test as it's easy to accidentally manually poke at methods in subtly different ways than TaskRunner will actually call them.